validator-network/src/network_impl.rs

```diff
@@ -199,7 +212,12 @@ where
         if let Ok(peer_id) = self.get_validator_peer_id(validator_id).await {
             self.network
                 .message(msg, peer_id)
-                .map_err(NetworkError::Request)
+                .map_err(|e| {
```

Comment:
Would a distinction between different kind of errors be feasible?

Reply:
It looks to me like a validator that changed its peer id should result in it not being dial-able anymore, thus resulting in a libp2p `OutboundFailure::DialFailure`, which we convert into a `RequestError::OutboundRequest(OutboundRequestError::DialFailure)`. However the network implementation isn't trivial and I'm worried that a changed peer id might also result in other errors in some cases or after code adjustments.

I'd lean towards clearing the cache on any error. That will result in some false-positives but also might be a more robust implementation.

Reply:
~It looks to me like a validator that changed its peer id should result in it not being dial-able anymore, thus resulting in a libp2p `OutboundFailure::DialFailure`, which we convert into a `RequestError::OutboundRequest(OutboundRequestError::DialFailure)`.~ The scenario currently results in a `InboundRequestError::NoReceiver`. However the network implementation isn't trivial and I'm worried that a changed peer id might also result in other errors in some cases or after code adjustments.

I'd lean towards clearing the cache on any error. That will result in some false-positives but also might be a more robust implementation.

Reply:
Sounds good.